EIPS/eip-2315.md

```diff
@@ -13,55 +13,67 @@ requires: 3540, 3670, 4200
 ## Abstract
 
-This proposal provides a _complete_, _safe_ and _efficient_ control-flow facility.
+This proposal provides a _complete_, _static_, _safe_ and _efficient_ control-flow facility.
 
 It deprecates `JUMP` and `JUMPI`.
 
-It introduces two opcodes to support simple subroutines:
-* `RJUMPSUB destination`— call a subroutine
-* `RETURNSUB` — return from call
-
 It depends on the two opcodes proposed by [EIP-4200](./eip-4200.md):
-* `RJUMP destination` — relative jump
-* `RJUMPI destination` — conditional relative jump
+* `RJUMP relative_offset` — relative jump to `PC + relative_offset`
+* `RJUMPI relative_offset` — conditional relative jump
+
+It introduces two opcodes to support calling and returning from subroutines:
+* `RJUMPSUB relative_offset` -- relative jump to subroutine
+* `RETURNSUB` -- return to `PC` after most recent `RJUMPSUB`.
 
-In concert with [EIP-3540](./eip-4200.md) and [EIP-3670](./eip-3670.md) it ensures that valid code will not execute invalid instructions, jump to invalid locations, or underflow stack. It will maintain consistent numbers of inputs and outputs for subroutines, and ensure that stack height is bounded in the absence of recursion.
+In concert with [EIP-3540](./eip-4200.md) and [EIP-3670](./eip-3670.md) it ensures, at initialization time, that valid code will not execute invalid instructions or jump to invalid locations, will not underflow stack, will maintain consistent numbers of inputs and outputs for subroutines, and will have bounded stack height in the absence of recursion.
 
-We believe this to be one of the simplest possible proposals that meets these requirements
+Valid code will support streaming, one-pass, and other near-linear compilers.
+
+This is among the simplest possible proposals that meets these requirements.
 
 ## Motivation
 
-### We want a complete control-flow facility.
+### We need a complete control-flow facility.
 
 Jumps, conditional jumps and subroutines were proposed by Alan Turing in 1945 as a means of organizing the logic of the code and the design of the memory crystals for his Automatic Computing Engine:
 > "We wish to be able to arrange that sequences of orders can divide at various points, continuing in different ways according to the outcome of the calculations to date...  We also wish to be able to arrange for the splitting up of operations into subsidiary operations... To start on a subsidiary operation we need only make a note of where we left off the major operation and then apply the first instruction of the subsidiary.  When the subsidiary is over we look up the note and continue with the major operation."
 > 
-> — Alan Turing — *cf. B.E. Carpenter, R.W. Doran, "The other Turing machine." The Computer Journal, Volume 20, Issue 3, January 1977.*
+> — Alan Turing — in [B.E. Carpenter, R.W. Doran, "The other Turing machine." The Computer Journal, Volume 20, Issue 3, DOI: 10.1093/comjnl/20.3.269, January 1977.](https://academic.oup.com/comjnl/article/20/3/269/751954)
```

Comment:
I strongly support including this reference link

Reply:
Yep.  This will do until we have consensus on that.